cmd/dms-uploader/dms-uploader.go

```diff
@@ -57,28 +55,29 @@ func run() int {
 	}
 
 	if *uri == "" {
-		log.Fatal("Target URI is not specified. See -h for usage.")
+		log.Fatal("Object storage URI is not specified. See -h for usage.")
 	}
 
-	handler, err := handlers.AvailableHandlers.Get(*uri)
-	if err != nil {
-		log.Fatal(err)
+	if *key == "" {
+		log.Fatal("Object storage key is not specified. See -h for usage.")
 	}
 
-	err = handler.NewSession(*id, *secret, *param)
+	storageDriver, err := drivers.ParseOSURL(*uri, false)
+	// path is passed along with the key when uploading
+	session := storageDriver.NewSession("")
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	ctx, cancelFn := context.WithTimeout(context.Background(), time.Hour)
 	defer cancelFn()
 
-	resUri, err := handler.Upload(ctx, *uri, os.Stdin)
+	resKey, err := session.SaveData(ctx, *key, os.Stdin, nil, time.Second*2)
```

Comment:
Added a `timeout` argument and set it to 10 sec by default. We would expect average upload time to be less than segment duration, and segments are normally 2 seconds long.

Reply:
Ah ok! I didn't think this would be used for uploading segments as well. For the entire MP4 files we'll likely need to configure the longer timeouts I guess.